SimpleChat.Dal/Interfaces/IRepository.cs

```diff
@@ -0,0 +1,24 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+
+namespace SimpleChat.Dal.Interfaces
+{
+    public interface IRepository<T> where T : class
+    {
+        IAsyncEnumerable<T> GetAllAsync();
```

Comment:
I don't think we really need an IAsyncEnumerable. Just an IEnumerable with pagination